<center>
   <h1> Amazon Sales Dataset - Sentiment Analysis </h1>
    <img src="./Assets/Amazon.png"></img>
<br><br>
</center>



<b> About Dataset </b>
This dataset is having the data of 1K+ Amazon Product's Ratings and Reviews as per their details listed on the official website of Amazon
<br>
<b> Features availible in the dataset :</b>
<br>
- product_id - Product ID
- product_name - Name of the Product
- category - Category of the Product
- discounted_price - Discounted Price of the Product
- actual_price - Actual Price of the Product
- discount_percentage - Percentage of Discount for the Product
- rating - Rating of the Product
- rating_count - Number of people who voted for the Amazon rating
- about_product - Description about the Product
- user_id - ID of the user who wrote review for the Product
- user_name - Name of the user who wrote review for the Product
- review_id - ID of the user review
- review_title - Short review
- review_content - Long review
- img_link - Image Link of the Product
- product_link - Official Website Link of the Product



<p><b>Inspiration about the dataset :</b>
Amazon is an American Tech Multi-National Company whose business interests include E-commerce, where they buy and store the inventory, and take care of everything from shipping and pricing to customer service and returns. I've created this dataset so that people can play with this dataset and do a lot of things as mentioned below :</p>

- Dataset Walkthrough
- Understanding Dataset Hierarchy
- Data Preprocessing
- Exploratory Data Analysis"
- Data Visualization
- Making Recommendation System
- 
This is a list of some of that things that you can do on this dataset. It's not definitely limited to the one that is mentioned there but a lot more other things can also be done.

In [246]:
# Importing the necessary libraries
import pandas as pd
import numpy as np

<b> Data Retrieval </b>

In [247]:
data = pd.read_csv("./Data/amazon.csv")

<b> Data Introduction </b>

In [248]:
data.head(10)

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   
5  B08Y1TFSP6  pTron Solero TB301 3A Type-C Data and Fast Cha...   
6  B08WRWPM22  boAt Micro USB 55 Tangle-free, Sturdy Micro US...   
7  B08DDRGWTJ             MI Usb Type-C Cable Smartphone (Black)   
8  B008IFXQFU  TP-Link USB WiFi Adapter for PC(TL-WN725N), N1...   
9  B082LZGK39  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   

                                            category discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             ₹399   
1  Computers&Accessories|Accessories&Peripherals|...             ₹199   
2  Computers&Accessories|Accessories&Peripherals|...             ₹199   
3  Computers&Accessories|Accessories&Peripherals|...             ₹329   
4  Computers&Accessories|Accessories&Peripherals|...             ₹154   
5  Computers&Accessories|Accessories&Peripherals|...             ₹149   
6  Computers&Accessories|Accessories&Peripherals|...          ₹176.63   
7  Computers&Accessories|Accessories&Peripherals|...             ₹229   
8  Computers&Accessories|NetworkingDevices|Networ...             ₹499   
9  Computers&Accessories|Accessories&Peripherals|...             ₹199   

  actual_price discount_percentage rating rating_count  \
0       ₹1,099                 64%    4.2       24,269   
1         ₹349                 43%    4.0       43,994   
2       ₹1,899                 90%    3.9        7,928   
3         ₹699                 53%    4.2       94,363   
4         ₹399                 61%    4.2       16,905   
5       ₹1,000                 85%    3.9       24,871   
6         ₹499                 65%    4.1       15,188   
7         ₹299                 23%    4.3       30,411   
8         ₹999                 50%    4.2     1,79,691   
9         ₹299                 33%    4.0       43,994   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   
5  Fast Charging & Data Sync: Solero TB301 Type-C...   
6  It Ensures High Speed Transmission And Chargin...   
7  1m long Type-C USB Cable|Sturdy and Durable. W...   
8  USB WiFi Adapter —— Speedy wireless transmissi...   
9  Universal Compatibility – It is compatible wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   
5  AEQ2YMXSZWEOHK2EHTNLOS56YTZQ,AGRVINWECNY7323CW...   
6  AG7C6DAADCTRQJG2BRS3RIKDT52Q,AFU7BOMPVJ7Q3TTA4...   
7  AHW6E5LQ2BDYOIVLAJGDH45J5V5Q,AF74RSGCHPZITVFSZ...   
8  AGV3IEFANZCKECFGUM42MRH5FNOA,AEBO7NWCNXKT4AESA...   
9  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   
5  Jayesh,Rajesh k.,Soopy,amazon customer,Aman,Sh...   
6  Vivek kumar,Amazon Customer,SARTHAK,Chiranjeev...   
7  Pavan A H,Jayesh bagad,Shridhar,rajendran,karu...   
8  Azhar JuMan,Anirudh Sood,Hari Krishnan PS,Akas...   
9  ArdKn,Nirbhay 

In [249]:
data.describe()

product_id                                       product_name  \
count         1465                                               1465   
unique        1351                                               1337   
top     B07JW9H4J1  Fire-Boltt Ninja Call Pro Plus 1.83" Smart Wat...   
freq             3                                                  5   

                                                 category discounted_price  \
count                                                1465             1465   
unique                                                211              550   
top     Computers&Accessories|Accessories&Peripherals|...             ₹199   
freq                                                  233               53   

       actual_price discount_percentage rating rating_count  \
count          1465                1465   1465         1463   
unique          449                  92     28         1143   
top            ₹999                 50%    4.1        9,378   
freq            120                  56    244            9   

                                            about_product  \
count                                                1465   
unique                                               1293   
top     [CHARGE & SYNC FUNCTION]- This cable comes wit...   
freq                                                    6   

                                                  user_id  \
count                                                1465   
unique                                               1194   
top     AHIKJUDTVJ4T6DV6IUGFYZ5LXMPA,AE55KTFVNXYFD5FPY...   
freq                                                   10   

                                                user_name  \
count                                                1465   
unique                                               1194   
top     $@|\|TO$|-|,Sethu madhav,Akash Thakur,Burger P...   
freq                                                   10   

                                                review_id  \
count                                                1465   
unique                                               1194   
top     R3F4T5TRYPTMIG,R3DQIEC603E7AY,R1O4Z15FD40PV5,R...   
freq                                                   10   

                                             review_title  \
count                                                1465   
unique                                               1194   
top     Worked on iPhone 7 and didn’t work on XR,Good ...   
freq                                                   10   

                                           review_content  \
count                                                1465   
unique                                               1212   
top     I am not big on camera usage, personally. I wa...   
freq                                                    8   

                                                 img_link  \
count                                                1465   
unique                                               1412   
top     https://m.media-amazon.com/images/I/413sCRKobN...   
freq                                                    3   

                                             product_link  
count                                                1465  
unique                                               1465  
top     https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
freq                                                    1

In [250]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1465 non-null   object
 1   product_name         1465 non-null   object
 2   category             1465 non-null   object
 3   discounted_price     1465 non-null   object
 4   actual_price         1465 non-null   object
 5   discount_percentage  1465 non-null   object
 6   rating               1465 non-null   object
 7   rating_count         1463 non-null   object
 8   about_product        1465 non-null   object
 9   user_id              1465 non-null   object
 10  user_name            1465 non-null   object
 11  review_id            1465 non-null   object
 12  review_title         1465 non-null   object
 13  review_content       1465 non-null   object
 14  img_link             1465 non-null   object
 15  product_link         1465 non-null   object
dtypes: obj

<b> Inferences : </b>
- Unique productID are 1351 and unique product name is 1337 ie. there are some false products
- ReviewID and ReviewTitle are same in unqiue number but the review content is more in number which means the short review and long review are different
- All data types are categorical

In [251]:
# Creating a list of columns 
column_list = data.columns.tolist()

In [252]:
column_list

['product_id',
 'product_name',
 'category',
 'discounted_price',
 'actual_price',
 'discount_percentage',
 'rating',
 'rating_count',
 'about_product',
 'user_id',
 'user_name',
 'review_id',
 'review_title',
 'review_content',
 'img_link',
 'product_link']

In [253]:
data.shape

(1465, 16)

In [254]:
data.head(1)

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   

                                            category discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             ₹399   

  actual_price discount_percentage rating rating_count  \
0       ₹1,099                 64%    4.2       24,269   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...

<b>Changes to perform :</b>
- Convert price to float
- Convert discount percentage to float
- Convert ratings to float
- Convert rating_count to float

In [255]:
# Converting price to integer

data['discounted_price'] = data['discounted_price'].str.replace("₹",'')
data['discounted_price'] = data['discounted_price'].str.replace(",",'')
data['discounted_price'] = data['discounted_price'].astype('float64')


In [256]:
data['actual_price'] = data['actual_price'].str.replace("₹",'')
data['actual_price'] = data['actual_price'].str.replace(",",'')
data['actual_price'] = data['actual_price'].astype('float64')

In [257]:
data[['actual_price','discounted_price']].head(10)

actual_price  discounted_price
0        1099.0            399.00
1         349.0            199.00
2        1899.0            199.00
3         699.0            329.00
4         399.0            154.00
5        1000.0            149.00
6         499.0            176.63
7         299.0            229.00
8         999.0            499.00
9         299.0            199.00

In [258]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1465 non-null   object 
 1   product_name         1465 non-null   object 
 2   category             1465 non-null   object 
 3   discounted_price     1465 non-null   float64
 4   actual_price         1465 non-null   float64
 5   discount_percentage  1465 non-null   object 
 6   rating               1465 non-null   object 
 7   rating_count         1463 non-null   object 
 8   about_product        1465 non-null   object 
 9   user_id              1465 non-null   object 
 10  user_name            1465 non-null   object 
 11  review_id            1465 non-null   object 
 12  review_title         1465 non-null   object 
 13  review_content       1465 non-null   object 
 14  img_link             1465 non-null   object 
 15  product_link         1465 non-null   o

In [259]:
data.category.value_counts()

category
Computers&Accessories|Accessories&Peripherals|Cables&Accessories|Cables|USBCables                                          233
Electronics|WearableTechnology|SmartWatches                                                                                 76
Electronics|Mobiles&Accessories|Smartphones&BasicMobiles|Smartphones                                                        68
Electronics|HomeTheater,TV&Video|Televisions|SmartTelevisions                                                               63
Electronics|Headphones,Earbuds&Accessories|Headphones|In-Ear                                                                52
                                                                                                                          ... 
Electronics|Cameras&Photography|Accessories|Batteries&Chargers|BatteryChargers                                               1
Computers&Accessories|NetworkingDevices|DataCards&Dongles                                             

In [260]:
data.rating.value_counts() # We tried to convert the rating to the float data type but it seems there is a categorical value in the field

rating
4.1    244
4.3    230
4.2    228
4.0    129
3.9    123
4.4    123
3.8     86
4.5     75
4       52
3.7     42
3.6     35
3.5     26
4.6     17
3.3     16
3.4     10
4.7      6
3.1      4
5.0      3
3.0      3
4.8      3
3.2      2
2.8      2
2.3      1
|        1
2        1
3        1
2.6      1
2.9      1
Name: count, dtype: int64

<b> Observations :</b>
- '|' is a value present in the ratings column which is categorical
- Category are divided by '|' 

In [261]:
data['rating'] = data['rating'].str.replace('|', '-1').astype('float64')

In [262]:
mean_value = np.mean(data['rating'])

In [263]:
data.loc[data['rating'] == -1, 'rating'] = mean_value

In [264]:
data.isna().sum()*100/len(data)

product_id             0.000000
product_name           0.000000
category               0.000000
discounted_price       0.000000
actual_price           0.000000
discount_percentage    0.000000
rating                 0.000000
rating_count           0.136519
about_product          0.000000
user_id                0.000000
user_name              0.000000
review_id              0.000000
review_title           0.000000
review_content         0.000000
img_link               0.000000
product_link           0.000000
dtype: float64

<b>Inference :</b> only rating_count has null values which is so small that it can be ignored 

In [265]:
# Splitting the categories

splitted = data['category'].str.split('|',expand=True)

In [266]:
splitted

0                           1  \
0     Computers&Accessories     Accessories&Peripherals   
1     Computers&Accessories     Accessories&Peripherals   
2     Computers&Accessories     Accessories&Peripherals   
3     Computers&Accessories     Accessories&Peripherals   
4     Computers&Accessories     Accessories&Peripherals   
...                     ...                         ...   
1460           Home&Kitchen      Kitchen&HomeAppliances   
1461           Home&Kitchen      Kitchen&HomeAppliances   
1462           Home&Kitchen  Heating,Cooling&AirQuality   
1463           Home&Kitchen  Heating,Cooling&AirQuality   
1464           Home&Kitchen      Kitchen&HomeAppliances   

                               2                         3          4     5  \
0             Cables&Accessories                    Cables  USBCables  None   
1             Cables&Accessories                    Cables  USBCables  None   
2             Cables&Accessories                    Cables  USBCables  None   
3             Cables&Accessories                    Cables  USBCables  None   
4             Cables&Accessories                    Cables  USBCables  None   
...                          ...                       ...        ...   ...   
1460  WaterPurifiers&Accessories  WaterPurifierAccessories       None  None   
1461      SmallKitchenAppliances         Rice&PastaCookers       None  None   
1462                 RoomHeaters            HeatConvectors       None  None   
1463                        Fans               ExhaustFans       None  None   
1464      SmallKitchenAppliances            SandwichMakers       None  None   

         6  
0     None  
1     None  
2     None  
3     None  
4     None  
...    ...  
1460  None  
1461  None  
1462  None  
1463  None  
1464  None  

[1465 rows x 7 columns]

In [267]:
#Renaming the categories in the column
splitted = splitted.rename(columns={0:'category_1', 1:'category_2', 2:'category_3'})

In [268]:
splitted.category_1.value_counts()

category_1
Electronics              526
Computers&Accessories    453
Home&Kitchen             448
OfficeProducts            31
MusicalInstruments         2
HomeImprovement            2
Toys&Games                 1
Car&Motorbike              1
Health&PersonalCare        1
Name: count, dtype: int64

In [269]:
splitted['category_1'] = splitted['category_1'].str.replace('&', ' & ')
splitted['category_1'] = splitted['category_1'].str.replace('OfficeProducts', 'Office Products')
splitted['category_1'] = splitted['category_1'].str.replace('MusicalInstruments', 'Musical Instruments')
splitted['category_1'] = splitted['category_1'].str.replace('HomeImprovement', 'Home Improvement')

In [270]:
splitted.category_1.value_counts()

category_1
Electronics                526
Computers & Accessories    453
Home & Kitchen             448
Office Products             31
Musical Instruments          2
Home Improvement             2
Toys & Games                 1
Car & Motorbike              1
Health & PersonalCare        1
Name: count, dtype: int64

In [271]:
splitted.category_2.value_counts()

category_2
Accessories&Peripherals                    381
Kitchen&HomeAppliances                     308
HomeTheater,TV&Video                       162
Mobiles&Accessories                        161
Heating,Cooling&AirQuality                 116
WearableTechnology                          76
Headphones,Earbuds&Accessories              66
NetworkingDevices                           34
OfficePaperProducts                         27
ExternalDevices&DataStorage                 18
Cameras&Photography                         16
HomeStorage&Organization                    16
HomeAudio                                   16
GeneralPurposeBatteries&BatteryChargers     14
Accessories                                 14
Printers,Inks&Accessories                   11
CraftMaterials                               7
Components                                   5
OfficeElectronics                            4
Electrical                                   2
Monitors                                     2
Mi

In [272]:
splitted['category_2'] = splitted['category_2'].str.replace('&', ' & ')
splitted['category_2'] = splitted['category_2'].str.replace(',', ', ')
splitted['category_2'] = splitted['category_2'].str.replace('HomeAppliances', 'Home Appliances')
splitted['category_2'] = splitted['category_2'].str.replace('AirQuality', 'Air Quality')
splitted['category_2'] = splitted['category_2'].str.replace('WearableTechnology', 'Wearable Technology')
splitted['category_2'] = splitted['category_2'].str.replace('NetworkingDevices', 'Networking Devices')
splitted['category_2'] = splitted['category_2'].str.replace('OfficePaperProducts', 'Office Paper Products')
splitted['category_2'] = splitted['category_2'].str.replace('ExternalDevices', 'External Devices')
splitted['category_2'] = splitted['category_2'].str.replace('DataStorage', 'Data Storage')
splitted['category_2'] = splitted['category_2'].str.replace('HomeStorage', 'Home Storage')
splitted['category_2'] = splitted['category_2'].str.replace('HomeAudio', 'Home Audio')
splitted['category_2'] = splitted['category_2'].str.replace('GeneralPurposeBatteries', 'General Purpose Batteries')
splitted['category_2'] = splitted['category_2'].str.replace('BatteryChargers', 'Battery Chargers')
splitted['category_2'] = splitted['category_2'].str.replace('CraftMaterials', 'Craft Materials')
splitted['category_2'] = splitted['category_2'].str.replace('OfficeElectronics', 'Office Electronics')
splitted['category_2'] = splitted['category_2'].str.replace('PowerAccessories', 'Power Accessories')
splitted['category_2'] = splitted['category_2'].str.replace('CarAccessories', 'Car Accessories')
splitted['category_2'] = splitted['category_2'].str.replace('HomeMedicalSupplies', 'Home Medical Supplies')
splitted['category_2'] = splitted['category_2'].str.replace('HomeTheater', 'Home Theater')
data['product_id'] = data['product_id'].str.strip()

Major Inferences and Necessary Features :

- product_ID
- product_name
- category
- discounted_price
- actual_price
- discount_percentage
- rating
- rating_count
- category_1
- category_2

In [273]:
df = data[['product_id', 'product_name', 'category', 'discounted_price', 'actual_price', 'discount_percentage', 'rating', 'rating_count']].copy()

In [274]:
df['category_1'] = splitted['category_1']
df['category_2'] = splitted['category_2']

In [276]:
df.head()

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category  discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             399.0   
1  Computers&Accessories|Accessories&Peripherals|...             199.0   
2  Computers&Accessories|Accessories&Peripherals|...             199.0   
3  Computers&Accessories|Accessories&Peripherals|...             329.0   
4  Computers&Accessories|Accessories&Peripherals|...             154.0   

   actual_price discount_percentage  rating rating_count  \
0        1099.0                 64%     4.2       24,269   
1         349.0                 43%     4.0       43,994   
2        1899.0                 90%     3.9        7,928   
3         699.0                 53%     4.2       94,363   
4         399.0                 61%     4.2       16,905   

                category_1                 category_2  
0  Computers & Accessories  Accessories & Peripherals  
1  Computers & Accessories  Accessories & Peripherals  
2  Computers & Accessories  Accessories & Peripherals  
3  Computers & Accessories  Accessories & Peripherals  
4  Computers & Accessories  Accessories & Peripherals

In [277]:
# dropping category now
df.drop(columns='category',inplace=True)

In [278]:
df.head()

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

   discounted_price  actual_price discount_percentage  rating rating_count  \
0             399.0        1099.0                 64%     4.2       24,269   
1             199.0         349.0                 43%     4.0       43,994   
2             199.0        1899.0                 90%     3.9        7,928   
3             329.0         699.0                 53%     4.2       94,363   
4             154.0         399.0                 61%     4.2       16,905   

                category_1                 category_2  
0  Computers & Accessories  Accessories & Peripherals  
1  Computers & Accessories  Accessories & Peripherals  
2  Computers & Accessories  Accessories & Peripherals  
3  Computers & Accessories  Accessories & Peripherals  
4  Computers & Accessories  Accessories & Peripherals

In [279]:
df.columns

Index(['product_id', 'product_name', 'discounted_price', 'actual_price',
       'discount_percentage', 'rating', 'rating_count', 'category_1',
       'category_2'],
      dtype='object')

In [280]:
# Creating codes for ratings 

rating_score = []
for score in df['rating']:
    if score<2.0 : rating_score.append('Poor')
    elif score<3.0 : rating_score.append('Below Average')
    elif score<4.0 : rating_score.append('Average')
    elif score<5.0 : rating_score.append('Above Average')
    else : rating_score.append('Excellent')

In [285]:
df['rating_score'] = rating_score

In [286]:
df.describe()

discounted_price   actual_price       rating
count       1465.000000    1465.000000  1465.000000
mean        3125.310874    5444.990635     4.096582
std         6944.304394   10874.826864     0.291574
min           39.000000      39.000000     2.000000
25%          325.000000     800.000000     4.000000
50%          799.000000    1650.000000     4.100000
75%         1999.000000    4295.000000     4.300000
max        77990.000000  139900.000000     5.000000

In [288]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   product_id           1465 non-null   object 
 1   product_name         1465 non-null   object 
 2   discounted_price     1465 non-null   float64
 3   actual_price         1465 non-null   float64
 4   discount_percentage  1465 non-null   object 
 5   rating               1465 non-null   float64
 6   rating_count         1463 non-null   object 
 7   category_1           1465 non-null   object 
 8   category_2           1465 non-null   object 
 9   rating_score         1465 non-null   object 
dtypes: float64(3), object(7)
memory usage: 114.6+ KB


In [289]:
df['rating_score'] = df['rating_score'].astype('category')

In [293]:
df['rating_score'] = df['rating_score'].cat.reorder_categories(['Below Average', 'Average', 'Above Average', 'Excellent'], ordered=True)

In [294]:
#Creating Difference of Price Column between Actual Price and Discounted Price

df['difference_price'] = df['actual_price'] - df['discounted_price']

In [295]:
df.head(4)

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   

   discounted_price  actual_price discount_percentage  rating rating_count  \
0             399.0        1099.0                 64%     4.2       24,269   
1             199.0         349.0                 43%     4.0       43,994   
2             199.0        1899.0                 90%     3.9        7,928   
3             329.0         699.0                 53%     4.2       94,363   

                category_1                 category_2   rating_score  \
0  Computers & Accessories  Accessories & Peripherals  Above Average   
1  Computers & Accessories  Accessories & Peripherals  Above Average   
2  Computers & Accessories  Accessories & Peripherals        Average   
3  Computers & Accessories  Accessories & Peripherals  Above Average   

   difference_price  
0             700.0  
1             150.0  
2            1700.0  
3             370.0

In [297]:
reviewers = data[['user_id','user_name']]
reviewers

user_id  \
0     AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1     AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2     AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3     AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4     AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   
...                                                 ...   
1460  AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...   
1461  AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...   
1462  AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...   
1463  AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...   
1464  AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...   

                                              user_name  
0     Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...  
1     ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...  
2     Kunal,Himanshu,viswanath,sai niharka,saqib mal...  
3     Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...  
4     rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...  
...                                                 ...  
1460  Prabha ds,Raghuram bk,Real Deal,Amazon Custome...  
1461  Manu Bhai,Naveenpittu,Evatira Sangma,JAGANNADH...  
1462  Nehal Desai,Danish Parwez,Amazon Customer,Amaz...  
1463  Shubham Dubey,E.GURUBARAN,Mayank S.,eusuf khan...  
1464  Rajib,Ajay B,Vikas Kahol,PARDEEP,Anindya Prama...  

[1465 rows x 2 columns]

In [298]:
#Splitting the strings in user_id column

reviewer_id_split = reviewers['user_id'].str.split(',', expand=False)

reviewer_id_split

0       [AG3D6O4STAQKAY2UVGEUV46KN35Q, AHMY5CWJMMK5BJR...
1       [AECPFYFQVRUWC3KGNLJIOREFP5LQ, AGYYVPDD7YG7FYN...
2       [AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA, AESFLDV2PT363T2...
3       [AEWAZDZZJLQUYVOVGBEUKSLXHQ5A, AG5HTSFRRE6NL3M...
4       [AE3Q6KSUK5P75D5HFYHCRAOLODSA, AFUGIFH5ZAFXRDS...
                              ...                        
1460    [AHITFY6AHALOFOHOZEOC6XBP4FEA, AFRABBODZJZQB6Z...
1461    [AFG5FM3NEMOL6BNFRV2NK5FNJCHQ, AGEINTRN6Z563RM...
1462    [AGVPWCMAHYQWJOQKMUJN4DW3KM5Q, AF4Q3E66MY4SR7Y...
1463    [AF2JQCLSCY3QJATWUNNHUSVUPNQQ, AFDMLUXC5LS5RXD...
1464    [AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA, AG7QNJ2SCS5VS5V...
Name: user_id, Length: 1465, dtype: object

In [299]:
#Making user id display 1 id per row

reviewer_id_exp = reviewer_id_split.explode()

reviewer_id_clean = reviewer_id_exp.reset_index(drop=True)

reviewer_id_clean

0        AG3D6O4STAQKAY2UVGEUV46KN35Q
1        AHMY5CWJMMK5BJRBBSNLYT3ONILA
2        AHCTC6ULH4XB6YHDY6PCH2R772LQ
3        AGYHHIERNXKA6P5T7CZLXKVPT7IQ
4        AG4OGOFWXJZTQ2HKYIOCOY3KXF2Q
                     ...             
11498    AHXCDNSXAESERITAFELQABFVNLCA
11499    AGRZD6CHLCUNOLMMIMIHUCG7PIFA
11500    AFQZVGSOSOJHKFQQMCEI4725QEKQ
11501    AEALVGXXIP46OZVXKRUXSDWZJMEA
11502    AGEFL3AY7YXEFZA4ZJU3LP7K7OJQ
Name: user_id, Length: 11503, dtype: object

In [300]:
#Splitting the strings in user_name column

reviewer_name_split = reviewers['user_name'].str.split(',', expand=False)

reviewer_name_split

0       [Manav, Adarsh gupta, Sundeep, S.Sayeed Ahmed,...
1       [ArdKn, Nirbhay kumar, Sagar Viswanathan, Asp,...
2       [Kunal, Himanshu, viswanath, sai niharka, saqi...
3       [Omkar dhale, JD, HEMALATHA, Ajwadh a., amar s...
4       [rahuls6099, Swasat Borah, Ajay Wadke, Pranali...
                              ...                        
1460    [Prabha ds, Raghuram bk, Real Deal, Amazon Cus...
1461    [Manu Bhai, Naveenpittu, Evatira Sangma, JAGAN...
1462    [Nehal Desai, Danish Parwez, Amazon Customer, ...
1463    [Shubham Dubey, E.GURUBARAN, Mayank S., eusuf ...
1464    [Rajib, Ajay B, Vikas Kahol, PARDEEP, Anindya ...
Name: user_name, Length: 1465, dtype: object

In [301]:
#Making user name display 1 id per row

review_name_exp = reviewer_name_split.explode()

reviewer_name_clean = review_name_exp.reset_index(drop=True)

reviewer_name_clean

0                   Manav
1            Adarsh gupta
2                 Sundeep
3          S.Sayeed Ahmed
4          jaspreet singh
               ...       
11510             PARDEEP
11511    Anindya Pramanik
11512         Vikas Singh
11513     Harshada Pimple
11514              Saw a.
Name: user_name, Length: 11515, dtype: object

In [302]:
#Creating 2 Data Frames to be merged

df21 = pd.DataFrame(data=reviewer_id_clean)
df22 = pd.DataFrame(data=reviewer_name_clean)

In [303]:
#Merging the 2 dataframe containing user_id and user_name

df2 = pd.merge(df21, df22, left_index=True, right_index=True)

In [304]:
#Final Check for the new dataframe

df2.head()

user_id       user_name
0  AG3D6O4STAQKAY2UVGEUV46KN35Q           Manav
1  AHMY5CWJMMK5BJRBBSNLYT3ONILA    Adarsh gupta
2  AHCTC6ULH4XB6YHDY6PCH2R772LQ         Sundeep
3  AGYHHIERNXKA6P5T7CZLXKVPT7IQ  S.Sayeed Ahmed
4  AG4OGOFWXJZTQ2HKYIOCOY3KXF2Q  jaspreet singh

In [307]:
#Exporting the data

df.to_csv("./Data/Cleaned.csv")

In [309]:
df2.to_csv("./Data/UserData.csv")